programming/robot_api/index.md

```diff
@@ -3,126 +3,119 @@ layout: page
 
-* [motors](/docs/programming/motors)
-* [power](/docs/programming/power)
-* [servos](/docs/programming/servos)
-* [ruggeduinos](/docs/programming/arduino/)
-* [vision](/docs/programming/vision/)
+<div class="info" markdown="1">
+Most examples in the documentation will assume you have created a `robot` object from the `Robot` class.
+If you see `robot` in a code example, it is assumed you have run the two lines above.
+</div>
 
-They can be used in your code just like the example above.
-Note that `motors`, `ruggeduinos`, and `servos` are Python lists, and so should be accessed as such.
-Here are some examples:
+Then, within your `robot` you can use the following attributes to access to the robot's hardware:
 
-~~~~~ python
-R.motor_board.motors[0].power = 0.5   # WILL work, if motor 0 exists
-R.motor_board.motors[1].power = -0.2  # WILL work, if motor 1 exists
-R.motor_board.motors.power = 0.42     # WON'T WORK
+* [kch]({{ site.baseurl }}/programming/leds)
+* [motor_board]({{ site.baseurl }}/programming/motors)
+* [power_board]({{ site.baseurl }}/programming/power)
+* [servo_board]({{ site.baseurl }}/programming/servos)
+* [arduino]({{ site.baseurl }}/programming/arduino/)
+* [camera]({{ site.baseurl }}/programming/vision/)
 
-# the above is similar to the situation for 'ruggeduinos' and 'servos'
-~~~~~
+The functions of each board are described on their respective pages.
 
-A number of examples in the documentation will assume you've instantiated the required `Robot` class and have called it `R`.
-From here in, if you see a `R.something`, the requirement of the `sr.robot3` import line and the instantiation of `Robot` as `R` is implicit.
 
-[Other Robot Attributes](#OtherRobotAttributes) {#OtherRobotAttributes}
-----------------------
+## Other Robot Attributes
 
-As well as the attributes listed above, the Robot class also has the following attributes, which you may find useful:
+As well as the attributes listed above, the `Robot` class also has the following attributes, which you may find useful:
 
 zone
-:    The number of the zone that the robot is associated with. Between `0` and `3`.
+:   The number of the scoring zone that the robot is associated with.
+    Between `0` and `3`.
+
+    This attribute is only available after the start button is pressed and will throw an error if accessed before.
+    See the [competition mode](./comp_mode) page for more information about this attribute.
 
 mode
 :   Whether the robot is running in competition mode.
-    When in a competition match, this will be `RobotMode.COMP`, and at all other times this will be `RobotMode.DEV`.
+    When in a competition match, this will be `COMP`, and at all other times this will be `DEV`.
+
+    This attribute is only available after the start button is pressed and will throw an error if accessed before.
+    See the [competition mode](./comp_mode) page for more information about this attribute.
 
     ~~~~~ python
     from sr.robot3 import *
```

Comment:
Nit: Someone might accidentally read these as strings. If we explicitly import them, people know where they come from.

```suggestion
    from sr.robot3 import Robot, COMP, DEV
```

Reply:
Discussed, not going to do

Reply:
![](https://media1.tenor.com/images/17a17ae6b93faf667b39af6d8fe34d68/tenor.gif?itemid=14359850)